File: trix/simplified/exercise.py
from trix.models import Exercise, Topic
from devilry.simplified import (FieldSpec, simplified_modelapi, SimplifiedModelApi, FilterSpec, FilterSpecs)
from authorization import AuthorizationMixin
from fakefieldcleaner import fakefieldcleaner

@fakefieldcleaner
@simplified_modelapi
class SimplifiedExercise(AuthorizationMixin):
    """ Simplified wrapper for :class:`trix.models.Exercise`. """
    class Meta(object):
        model = Exercise
        searchfields = FieldSpec('short_name', 'long_name', 'topics__name')
        resultfields = FieldSpec('id', 'short_name', 'long_name', 'text', 'points',
                                 'topics__id', 'prerequisites__id',
                                 topics=['topics__name'], prerequisites=['prerequisites__name'])

#       editablefields = ('id', 'short_name', 'long_name', 'text', 'points')
        fake_editablefields = ('fake_topics', 'fake_prerequisites')
        fake_real_map =  {'fake_topics': ('topics', Topic, 'pk', 'name'),
                          'fake_prerequisites': ('prerequisites', Topic, 'pk', 'name')}
        filters = FilterSpecs(FilterSpec('topics__id'))
        methods = ['create', 'read', 'update', 'delete', 'search']

File: trix/static/extjstrix/Exercise.js
Ext.define('trix.Exercise', {
    extend: 'Ext.data.Model',
    requires: ['trix_extjshelpers.RestProxy'],
    fields: [
        {"type": "int", "name": "id"},
        {"type": "auto", "name": "short_name"},
        {"type": "auto", "name": "long_name"},
        {"type": "auto", "name": "text"},
        {"type": "int", "name": "points"},
        {"type": "auto", "name": "topics__id"},
        {"type": "auto", "name": "prerequisites__id"},
        {"type": "auto", "name": "topics__name"},
        {"type": "auto", "name": "prerequisites__name"},
        {"type": "auto", "name": "fake_topics"},
        {"type": "auto", "name": "fake_prerequisites"}],
    idProperty: 'id',
    proxy: Ext.create('trix_extjshelpers.RestProxy', {
        url: '/trix/restfulsimplifiedexercise/',
        extraParams: {
            getdata_in_qrystring: true,
            result_fieldgroups: '["topics", "prerequisites"]'
        },

        reader: {
            type: 'json',
            root: 'items',
            totalProperty: 'total'
        },

        writer: {
            type: 'json'
        }
    })
});
